crates/context_aware_config/src/api/config/handlers.rs

```diff
@@ -180,10 +181,9 @@ pub fn generate_config_from_version(
         {
             Ok((latest_version, config)) => {
                 *version = Some(latest_version);
-
-                serde_json::from_value::<Config>(config).map_err(|err| {
+                serde_json::from_value::<Config>(config).or_else(|err| {
                     log::error!("failed to decode config: {}", err);
-                    unexpected_error!("failed to decode config")
+                    generate_cac(conn)
```

Comment:
`blocker`: we should fail here, rather than generate_cac. The user expects the CAC for the version they requested and may act based on that assumption. If we are sending the latest cac regardless of failure, they may act on the wrong assumptions.

Reply:
@Datron , generate_cac is in else case where user is not providing the version

Reply:
Hmm okay, I will create a ticket for this and we can remove both later